### PR TITLE
Port diabetic_retinopathy task from OATML/bdl-benchmarks/diabetic_retninopathy*

### PR DESCRIPTION
**Pull request by filangel**
- [x] Add Kaggle's diabetic-retinopathy-detection dataset, with `btgraham-300` preprocessing.
- [ ] Add implementations for [baselines](https://github.com/OATML/bdl-benchmarks/tree/alpha/baselines/diabetic_retinopathy_diagnosis).

### CODE REVIEW

File: baselines/diabetic_retinopathy_detection/deterministic.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+import os
+import time
+
+from absl import app
+from absl import flags
+from absl import logging
+
+import tensorflow as tf
+import uncertainty_baselines as ub
+import utils  # local file import
```
Comment (znado) [inline baselines/diabetic_retinopathy_detection/deterministic.py:27]: is it possible to change this to `from uncertainty_baselines.baselines.diabetic_retinopathy_detection import utils`?

File: uncertainty_baselines/datasets/base.py [hunk 2]
```diff
@@ -34,12 +34,12 @@ class OodSplit(enum.Enum):
...
 
+
 # For datasets like UCI, the tf.data.Dataset returned by _read_examples will
 # have elements that are Sequence[tf.Tensor], but for TFDS datasets they will be
 # Dict[str, tf.Tensor].
-PreProcessFn = Callable[
-    [Union[int, Sequence[tf.Tensor], Dict[str, tf.Tensor]]],
-    Dict[str, tf.Tensor]]
+PreProcessFn = Callable[[Union[int, Sequence[tf.Tensor], Dict[str, tf.Tensor]]],
+                        Dict[str, tf.Tensor]]
```
Comment (znado) [inline uncertainty_baselines/datasets/base.py:42]: can we revert these edits (maybe an auto-linter caused them?)

File: uncertainty_baselines/datasets/datasets.py [hunk 3]
```diff
@@ -98,13 +98,13 @@ def get(
   logging.info(
       'Building dataset %s with additional kwargs:\n%s',
       dataset_name,
-      json.dumps(hyperparameters, indent=2, sort_keys=True))
+      json.dumps(hyperparameters, indent=2, sort_keys=True),
+  )
```
Comment (znado) [inline uncertainty_baselines/datasets/datasets.py:102]: we've generally avoided the hanging parentheses in the codebase, the other lint edits are fine (we haven't been super consistent on how to do indentations for vertical parameter/arg lists)

**Review by znado — APPROVED** [review]
looks fantastic :)

File: baselines/diabetic_retinopathy_detection/deterministic.py (hunk 1 above)
Comment (nband) [reply baselines/diabetic_retinopathy_detection/deterministic.py:27]: Is uncertainty_baselines.baselines supposed to be available as a module? Other imports in the repo seem to only access the uncertainty_baselines subdirectory, and the ['datasets', 'models', 'optimizers', 'strategy_utils', 'utils'] modules

**Comment by nband** [comment]
@googlebot I fixed it.

**Comment by nband** [comment]
@googlebot I consent.

**Pull request merged**